### PR TITLE
Fix some doc link

### PR DESCRIPTION
**Pull request by thiolliere**
This fix some of the dead link.

But there is still some like in client/network which I don't really know how to solve. Also this is a bit a chicken egg problem, as long as we don't fix all error then we can't put it on CI, but as long as we don't put it on CI we keep adding some error.

After this PR I'll probably just remove all the dead link and open an issue linking to all the removal so people can fix them afterwards.

### CODE REVIEW

File: frame/beefy-mmr/primitives/src/lib.rs [hunk 1]
```diff
@@ -44,7 +44,7 @@ pub type Hash = [u8; 32];
 /// Generic hasher trait.
 ///
 /// Implement the function to support custom way of hashing data.
-/// The implementation must return a [Hash] type, so only 32-byte output hashes are supported.
+/// The implementation must return a [Hash](type@Hash) type, so only 32-byte output hashes are supported.
```
Comment (kianenigma) [inline frame/beefy-mmr/primitives/src/lib.rs:47]: what's this syntax?

**Review by kianenigma — APPROVED** [review]
Really up for making this checked in the CI

File: frame/beefy-mmr/primitives/src/lib.rs (hunk 1 above)
Comment (thiolliere) [reply frame/beefy-mmr/primitives/src/lib.rs:47]: the type is ambiguous, with the Hash derive macro IIRC, and the compiler suggested this to disambiguate.

Comment (bkchr) [reply frame/beefy-mmr/primitives/src/lib.rs:47]: https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html#namespaces-and-disambiguators

**Review by bkchr — APPROVED** [review]

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**